files/elb_lookup.py

```diff
@@ -1,60 +1,99 @@
 import os
 import boto3
 
-client = boto3.client('elb')
-clientv2 = boto3.client('elbv2')
-vpc_id = os.environ['VPC_ID']
-cluster_name = os.environ['CLUSTER_NAME']
+def _generate_lb_name_and_tags(client=None,
+                               clientv2=None):
+    ''' Generate tuples (LB name, tag description)
+    for both LB api versions, interleaving queries
+    between the v1 and v2 APIs.
+    '''
 
-def _get_elb_name():
+    vpc_id = os.environ['VPC_ID']
+
+    # Get the first page of results from both
+    # ELBv1 and ELBv2 APIs
     response = client.describe_load_balancers()
     response2 = clientv2.describe_load_balancers()
     load_balancers = response['LoadBalancerDescriptions']
     nlb_load_balancers = response2['LoadBalancers']
 
-    # get all the ELBs in this VPC
-    load_balancer_names_in_vpc = []
-    for lb in load_balancers:
-        if lb['VPCId'] == vpc_id:
-            load_balancer_names_in_vpc.append(lb["LoadBalancerName"])
-
-    # get all the ELBs in this VPC
-    nlb_load_balancer_arns_in_vpc = []
-    for lb in nlb_load_balancers:
-        if lb['VpcId'] == vpc_id:
-            nlb_load_balancer_arns_in_vpc.append(lb["LoadBalancerArn"])
-
-    tag_descriptions = []
-    # get the tags
-    if len(load_balancer_names_in_vpc) > 0:
-        response = client.describe_tags(
-            LoadBalancerNames=load_balancer_names_in_vpc)
-        tag_descriptions += response['TagDescriptions']
-    if len(nlb_load_balancer_arns_in_vpc) > 0:
-        response2 = clientv2.describe_tags(
-            ResourceArns=nlb_load_balancer_arns_in_vpc)
-        tag_descriptions += response2['TagDescriptions']
-
-    # return the name of the ELB corresponding to this cluster
-    for description in tag_descriptions:
+    v1_next_token = response.get('NextMarker')
+    v2_next_token = response2.get('NextMarker')
+
+    # do-while loop, conditional at bottom
+    while True:
+
+        # get all the ELBs in this VPC
+        load_balancer_names_in_vpc = []
+        for lb in load_balancers:
+            if lb['VPCId'] == vpc_id:
+                load_balancer_names_in_vpc.append(lb["LoadBalancerName"])
+
+        # get all the ELBs in this VPC
+        nlb_load_balancer_arns_in_vpc = []
+        for lb in nlb_load_balancers:
+            if lb['VpcId'] == vpc_id:
+                nlb_load_balancer_arns_in_vpc.append(lb["LoadBalancerArn"])
```

Comment:
we are only considering LBs in the expected VPC. the key is named differently between the two APIs.